src/main/java/org/gridsuite/study/server/RestResponseEntityExceptionHandler.java

```diff
@@ -19,36 +21,45 @@
-                return ResponseEntity.status(HttpStatus.FORBIDDEN).body(LOADFLOW_NOT_RUNNABLE);
-            case LOADFLOW_RUNNING:
-                return ResponseEntity.status(HttpStatus.FORBIDDEN).body(LOADFLOW_RUNNING);
-            case SECURITY_ANALYSIS_RUNNING:
-                return ResponseEntity.status(HttpStatus.FORBIDDEN).body(SECURITY_ANALYSIS_RUNNING);
-            case NOT_ALLOWED:
-            case CANT_DELETE_ROOT_NODE:
-                return ResponseEntity.status(HttpStatus.FORBIDDEN).body(NOT_ALLOWED);
-            case LINE_MODIFICATION_FAILED:
-            case LOAD_CREATION_FAILED:
-            case GENERATOR_CREATION_FAILED:
-            case LINE_CREATION_FAILED:
-                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(studyException.getMessage());
-            case DELETE_EQUIPMENT_FAILED:
-                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(studyException.getMessage());
-            default:
-                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        if (exception instanceof StudyException) {
+            StudyException studyException = (StudyException) exception;
+            switch (studyException.getType()) {
+                case ELEMENT_NOT_FOUND:
+                case STUDY_NOT_FOUND:
+                case SECURITY_ANALYSIS_NOT_FOUND:
+                    return ResponseEntity.status(HttpStatus.NOT_FOUND).body(studyException.getType());
+                case CASE_NOT_FOUND:
+                    return ResponseEntity.status(HttpStatus.FAILED_DEPENDENCY).body(CASE_NOT_FOUND);
+                case STUDY_ALREADY_EXISTS:
+                    return ResponseEntity.status(HttpStatus.CONFLICT).body(STUDY_ALREADY_EXISTS);
+                case LOADFLOW_NOT_RUNNABLE:
+                    return ResponseEntity.status(HttpStatus.FORBIDDEN).body(LOADFLOW_NOT_RUNNABLE);
+                case LOADFLOW_RUNNING:
+                    return ResponseEntity.status(HttpStatus.FORBIDDEN).body(LOADFLOW_RUNNING);
+                case SECURITY_ANALYSIS_RUNNING:
+                    return ResponseEntity.status(HttpStatus.FORBIDDEN).body(SECURITY_ANALYSIS_RUNNING);
+                case NOT_ALLOWED:
+                case CANT_DELETE_ROOT_NODE:
+                    return ResponseEntity.status(HttpStatus.FORBIDDEN).body(NOT_ALLOWED);
+                case LINE_MODIFICATION_FAILED:
+                case LOAD_CREATION_FAILED:
+                case GENERATOR_CREATION_FAILED:
+                case LINE_CREATION_FAILED:
+                    return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(studyException.getMessage());
+                case DELETE_EQUIPMENT_FAILED:
+                    return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(studyException.getMessage());
+                default:
+            }
+        } else if (exception instanceof ServerWebInputException) {
```

Comment:
I think it's too complicated 
I think is better to do the conversion in the service in order to facilitate reading